### ISSUE TITLE
Searching in a long line can be slow

### ISSUE THREAD

**Issue by gwsw** [gwsw/less#370]
Searching for text near the end of a very long line can be unreasonably slow. For example, run
```
seq 30000 | tr '\n' ' ' | less
```
This produces a file containing one line which is 168,894 chars long. Type ```/29999``` to search for a string near the end of the line. On a fairly fast desktop system this takes 160 seconds to complete.

**Comment by gwsw** [comment]
Fixed in 236e0419e86ec323ea6b10bbd22c67e7c6bd76bd.

**Comment by gwsw** [comment]
Note: 236e0419e86ec323ea6b10bbd22c67e7c6bd76bd was buggy when used with non-ASCII text. 
Really fixed in 3f8d4ab3bdc484493bbce61551afa45b94855824.

**Comment by mohd-akram** [comment]
This is still slow for me, but it was never 160 seconds. Performance is the same between versions (on a MBP 2016):

less 643:

```shell
$ less --version | head -n1
less 643 (POSIX regular expressions)
$ seq 30000 | tr '\n' ' ' | time less +/29999 +q
       10.80 real         7.49 user         0.12 sys
```

less 651x:

```shell
$ ./less --version | head -n1
less 651x (POSIX regular expressions)
$ seq 30000 | tr '\n' ' ' | time ./less +/29999 +q
        9.25 real         6.21 user         0.07 sys
```
